### ISSUE TITLE
ArgumentError on pre-joined query

### ISSUE THREAD

**Issue by TWeatherston** [elmkarami/sqlalchemy-filters-plus#19]
I'm not totally sure what's causing this so apologies for the somewhat vague issue. Hopefully you'll have a better idea!

I have a query with a couple of joins on it already and when I attempt to filter it, I'm getting the following exception:

```bash
  File "/usr/local/lib/python3.9/site-packages/sqlalchemy_filters/filters.py", line 563, in apply
    query = self._apply_join(query)
  File "/usr/local/lib/python3.9/site-packages/sqlalchemy_filters/filters.py", line 364, in _apply_join
    if is_already_joined(query, join[0]):
  File "/usr/local/lib/python3.9/site-packages/sqlalchemy_filters/utils.py", line 37, in is_already_joined
    return join_table in [_[0] for _ in query._legacy_setup_joins]
  File "/usr/local/lib/python3.9/site-packages/sqlalchemy/sql/operators.py", line 366, in __eq__
    return self.operate(eq, other)
  File "/usr/local/lib/python3.9/site-packages/sqlalchemy/orm/attributes.py", line 317, in operate
    return op(self.comparator, *other, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/sqlalchemy/orm/relationships.py", line 1302, in __eq__
    self.property._optimized_compare(
  File "/usr/local/lib/python3.9/site-packages/sqlalchemy/orm/relationships.py", line 1690, in _optimized_compare
    raise sa_exc.ArgumentError(
sqlalchemy.exc.ArgumentError: Mapped instance expected for relationship comparison to object.   Classes, queries and other SQL elements are not accepted in this context; for comparison with a subquery, use DataSource.owner.has(**criteria).
```

**Comment by elmkarami** [comment]
@TWeatherston could you please share the structure of your models/filters to have a full understanding of the issue. Thanks.

**Comment by TWeatherston** [comment]
Yep sure, this is how my models, filter and query are being used:

```python
class TableA(db.Model):
    col1 = Column(db.String(120), nullable=False)
    col2 = relationship("TableB", primaryjoin="foreign(TableA.col1) == TableB.col1")
    col3 = Column(db.String(80))
    col4 = relationship("TableC", primaryjoin="foreign(TableA.col3) == TableC.col1")


class TableB(db.Model):
    col1 = Column(db.String(120), nullable=False)
    col2 = Column(db.String(120), nullable=False)


class TableC(db.Model):
    col1 = Column(db.String(80), unique=True, nullable=False)
    col2 = Column(db.String(120), nullable=False)


class TableAFilter(Filter):
    field1 = Field(field_name="col4.col2")


query = TableA.query
query = query.join(TableA.col2, isouter=True).join(TableA.col4, isouter=True)
my_filter = TableAFilter(data={"field1": "value"}, query=query)
query = my_filter.apply()
```
I'm using flask-sqlalchemy v2.5.1

**Comment by elmkarami** [comment]
@TWeatherston the issue is fixed(version 1.1.4). thank you for reporting it.
